File: 38.count-and-say.py
class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        nums = [1]
        for _ in range(n - 1):
            nums2 = []
            i = 0
            while i < len(nums):
                j = 1;
                while i + 1 < len(nums) and nums[i] == nums[i + 1]:
                    j += 1
                    i += 1
                nums2.append(j)
                nums2.append(nums[i])
                i += 1
            nums = nums2
        return "".join(str(i) for i in nums)

File: 38.count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        vector<int> nums = { 1 };
        while (--n) {
            vector<int> nums2;
            for (int i = 0; i < nums.size(); i++) {
                int j = 1;
                while (i + 1 < nums.size() && nums[i] == nums[i + 1]) {
                    j++;
                    i++;
                }
                nums2.push_back(j);
                nums2.push_back(nums[i]);
            }
            nums = nums2;
        }
        string s;
        for (int n : nums) {
            s += ('0' + n);
        }
        return s;
    }
};